temp/legacy-code/src/main/java/com/ivy/legacy/domain/deprecated/logic/PreloadDataLogic.kt

```diff
@@ -171,17 +178,23 @@ class PreloadDataLogic @Inject constructor(
     )
 
     private suspend fun preloadCategory(
-        data: CreateCategoryData
+        data: CreateCategoryData,
     ) {
-        val category = Category(
-            name = data.name,
-            color = data.color.toArgb(),
-            icon = data.icon,
-            orderNum = categoryOrderNum++,
-            isSynced = false
-        )
+        val category: Category? = either {
+            Category(
+                name = NotBlankTrimmedString.from(data.name.trim()).bind(),
```

Comment:
no need for `.trim()` the `NotBlankTrimmedString` does it for us

Reply:
Updated